pineappl_capi/src/lib.rs

```diff
@@ -2273,3 +2274,24 @@ pub unsafe extern "C" fn pineappl_grid_evolve(
 
     Box::new(fk_table.unwrap().into_grid())
 }
+
+/// Optimize the size of this FK-table by removing heavy quark flavors assumed to be zero.
+///
+/// # Safety
+///
+/// If `grid` does not point to a valid `Grid` object, for example when `grid` is the null pointer,
+/// this function is not safe to call.
+///
+/// # Panics
+///
+/// This function panics if `grid` is not an FK table-like object.
+#[no_mangle]
+pub unsafe extern "C" fn pineappl_fktable_optimize(
+    grid: *mut Grid,
+    assumptions: FkAssumptions,
+) -> Box<FkTable> {
```

Comment:
If we return a new object we can leave the parameter as `*const Grid`, because we don't modify the `Grid` in that case. However, returning a new object is probably surprising since `pineappl_grid_optimize` doesn't do that. Instead I suggest the following: use [`ptr::replace`](https://doc.rust-lang.org/stable/std/ptr/fn.replace.html) once to 'steal' `grid` and replace it with a dummy grid, then convert the stolen grid to `FkTable` and subsequently perform the optimization, and finally perform the replacement in the opposite way to 'return' the grid via the original parameter - I hope that works.

Reply:
This is actually brilliant! Moving the pointer into a dummy grid was not needed but replacing it with the optimized grid works perfect 🔥

Reply:
Yes, that's right (I missed that)!

@vbertone would it be possible for you to upload the **unoptimized** FK table somewhere so I can test with that? It is a bit hard to check the memory differences with a few Mbs FK table. Thanks a lot!